pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/UpsertUtils.java

```diff
@@ -190,13 +191,14 @@ public Comparable getComparisonValue(int docId) {
 
       for (int i = 0; i < _comparisonColumnReaders.length; i++) {
         PinotSegmentColumnReader columnReader = _comparisonColumnReaders[i];
-        Comparable comparisonValue = (Comparable) UpsertUtils.getValue(columnReader, docId);
+        Comparable comparisonValue = null;
+        if (!columnReader.isNull(docId)) {
```

Comment:
I'm not sure if having this null-check is actually useful.  I think the logic within ComparisonColumns would stay the same regardless.

Reply:
We need this null check, or it won't work properly (we shouldn't rely on default null value to be smaller than actual value)